--- elements.c.orig	2008-06-07 20:54:46.000000000 +0300
+++ elements.c	2008-09-24 22:11:25.000000000 +0300
@@ -260,8 +260,8 @@
 	}
 	else
 	{
-		compLogMessage (display, "Elements", CompLogLevelWarn,
-			    "Not a valid element type");
+		//compLogMessage (display, "Elements", CompLogLevelWarn,
+		//	    "Not a valid element type");
 	}
 }
 
@@ -833,12 +833,12 @@
 					&eScreen->textu[count].height);
 		if (!eScreen->textu[count].loaded)
 		{
-		    compLogMessage (s->display, "Elements", CompLogLevelWarn,
-			    "Texture (Autumn) not found : %s", ed->texFiles[0][i].s);
+		    //compLogMessage (s->display, "Elements", CompLogLevelWarn,
+		//	    "Texture (Autumn) not found : %s", ed->texFiles[0][i].s);
 		    continue;
 		}
-		compLogMessage (s->display, "Elements", CompLogLevelInfo,
-			"Loaded Texture (Autumn)%s", ed->texFiles[0][i].s);
+		//compLogMessage (s->display, "Elements", CompLogLevelInfo,
+		//	"Loaded Texture (Autumn)%s", ed->texFiles[0][i].s);
 		}
 		mat = &eScreen->textu[count].tex.matrix;
 		aTex = &eScreen->textu[count];
@@ -879,12 +879,12 @@
 				&eScreen->textu[count].height);
 		if (!eScreen->textu[count].loaded)
 		{
-		    compLogMessage (s->display, "Elements", CompLogLevelWarn,
-			    "Texture (Firefly) not found : %s", ed->texFiles[1][i].s);
+		    //compLogMessage (s->display, "Elements", CompLogLevelWarn,
+		//	    "Texture (Firefly) not found : %s", ed->texFiles[1][i].s);
 		    continue;
 		}
-		compLogMessage (s->display, "Elements", CompLogLevelInfo,
-			"Loaded Texture (Firefly) %s", ed->texFiles[1][i].s);
+		//compLogMessage (s->display, "Elements", CompLogLevelInfo,
+		//	"Loaded Texture (Firefly) %s", ed->texFiles[1][i].s);
 		}
 		mat = &eScreen->textu[count].tex.matrix;
 		aTex = &eScreen->textu[count];
@@ -926,12 +926,12 @@
 				&eScreen->textu[count].height);
 		if (!eScreen->textu[count].loaded)
 		{
-			compLogMessage (s->display, "Elements", CompLogLevelWarn,
-			    "Texture (snow) not found : %s", ed->texFiles[2][i].s);
+			//compLogMessage (s->display, "Elements", CompLogLevelWarn,
+			//    "Texture (snow) not found : %s", ed->texFiles[2][i].s);
 			continue;
 		}
-		compLogMessage (s->display, "Elements", CompLogLevelInfo,
-			"Loaded Texture (snow) %s", ed->texFiles[2][i].s);
+		//compLogMessage (s->display, "Elements", CompLogLevelInfo,
+		//	"Loaded Texture (snow) %s", ed->texFiles[2][i].s);
 	}
 		mat = &eScreen->textu[count].tex.matrix;
 		aTex = &eScreen->textu[count];
@@ -973,12 +973,12 @@
 				&eScreen->textu[count].height);
 		if (!eScreen->textu[count].loaded)
 		{
-			compLogMessage (s->display, "Elements", CompLogLevelWarn,
-			    "Texture (stars) not found : %s", ed->texFiles[3][i].s);
+			//compLogMessage (s->display, "Elements", CompLogLevelWarn,
+			//    "Texture (stars) not found : %s", ed->texFiles[3][i].s);
 			continue;
 		}
-		compLogMessage (s->display, "Elements", CompLogLevelInfo,
-			"Loaded Texture (stars)%s", ed->texFiles[3][i].s);
+		//compLogMessage (s->display, "Elements", CompLogLevelInfo,
+		//	"Loaded Texture (stars)%s", ed->texFiles[3][i].s);
 	}
 		mat = &eScreen->textu[count].tex.matrix;
 		aTex = &eScreen->textu[count];
@@ -1020,12 +1020,12 @@
 				&eScreen->textu[count].height);
 		if (!eScreen->textu[count].loaded)
 		{
-			compLogMessage (s->display, "Elements", CompLogLevelWarn,
-			    "Texture (bubbles) not found : %s", ed->texFiles[4][i].s);
+			//compLogMessage (s->display, "Elements", CompLogLevelWarn,
+			//    "Texture (bubbles) not found : %s", ed->texFiles[4][i].s);
 			continue;
 		}
-		compLogMessage (s->display, "Elements", CompLogLevelInfo,
-			"Loaded Texture (bubbles)%s", ed->texFiles[4][i].s);
+		//compLogMessage (s->display, "Elements", CompLogLevelInfo,
+		//	"Loaded Texture (bubbles)%s", ed->texFiles[4][i].s);
 	}
 		mat = &eScreen->textu[count].tex.matrix;
 		aTex = &eScreen->textu[count];
@@ -1103,7 +1103,7 @@
 	setupDisplayList (eScreen);
 	WRAP (eScreen, s, paintOutput, elementsPaintOutput);
 	WRAP (eScreen, s, drawWindow, elementsDrawWindow);
-	eScreen->timeoutHandle = compAddTimeout (elementsGetUpdateDelay (s->display),
+	eScreen->timeoutHandle = compAddTimeout (elementsGetUpdateDelay (s->display),10,
 					stepPositions, s);
 
 	return TRUE;
@@ -1407,7 +1407,7 @@
 			if (eScreen->timeoutHandle)
 		  	  	compRemoveTimeout (eScreen->timeoutHandle);
 				eScreen->timeoutHandle =
-					compAddTimeout (elementsGetUpdateDelay (d),
+					compAddTimeout (elementsGetUpdateDelay (d),10,
 					stepPositions, s);
 			}
 		}
