--- src/base/ftlcdfil.c.orig	2010-10-09 14:26:07.721109451 +0200
+++ src/base/ftlcdfil.c	2010-10-09 14:27:12.235099307 +0200
@@ -288,7 +288,7 @@
     /* the values here sum up to a value larger than 256, */
     /* providing a cheap gamma correction                 */
     static const FT_Byte  default_filter[5] =
-                            { 0x10, 0x40, 0x70, 0x40, 0x10 };
+                            { 0x1c, 0x38, 0x60, 0x38, 0x1c };
 
 
     if ( !library )
