diff --binary -ruN codeblocks-8.02.orig/src/plugins/astyle/Makefile.am codeblocks-8.02/src/plugins/astyle/Makefile.am
--- codeblocks-8.02.orig/src/plugins/astyle/Makefile.am	2007-08-08 14:30:43.000000000 +0400
+++ codeblocks-8.02/src/plugins/astyle/Makefile.am	2008-03-02 11:41:24.000000000 +0300
@@ -5,9 +5,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libastyle.la
+plugins_LTLIBRARIES = libastyle.la
 
 libastyle_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/autosave/Makefile.am codeblocks-8.02/src/plugins/autosave/Makefile.am
--- codeblocks-8.02.orig/src/plugins/autosave/Makefile.am	2007-07-16 18:07:04.000000000 +0400
+++ codeblocks-8.02/src/plugins/autosave/Makefile.am	2008-03-02 11:41:34.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libautosave.la
+plugins_LTLIBRARIES = libautosave.la
 
 libautosave_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/classwizard/Makefile.am codeblocks-8.02/src/plugins/classwizard/Makefile.am
--- codeblocks-8.02.orig/src/plugins/classwizard/Makefile.am	2007-03-26 01:31:53.000000000 +0400
+++ codeblocks-8.02/src/plugins/classwizard/Makefile.am	2008-03-02 11:41:44.000000000 +0300
@@ -4,9 +4,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libclasswizard.la
+plugins_LTLIBRARIES = libclasswizard.la
 
 libclasswizard_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/codecompletion/Makefile.am codeblocks-8.02/src/plugins/codecompletion/Makefile.am
--- codeblocks-8.02.orig/src/plugins/codecompletion/Makefile.am	2007-03-26 01:31:53.000000000 +0400
+++ codeblocks-8.02/src/plugins/codecompletion/Makefile.am	2008-03-02 11:42:08.000000000 +0300
@@ -5,9 +5,9 @@
 		-I$(top_srcdir)/src/include/wxFlatNotebook/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libcodecompletion.la
+plugins_LTLIBRARIES = libcodecompletion.la
 
 libcodecompletion_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/compilergcc/Makefile.am codeblocks-8.02/src/plugins/compilergcc/Makefile.am
--- codeblocks-8.02.orig/src/plugins/compilergcc/Makefile.am	2007-09-11 01:05:41.000000000 +0400
+++ codeblocks-8.02/src/plugins/compilergcc/Makefile.am	2008-03-02 11:42:21.000000000 +0300
@@ -9,9 +9,9 @@
 
 AM_CXXFLAGS = -DDEPSLIB_UNIX
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libcompiler.la
+plugins_LTLIBRARIES = libcompiler.la
 
 libcompiler_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/AutoVersioning/Makefile.am codeblocks-8.02/src/plugins/contrib/AutoVersioning/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/AutoVersioning/Makefile.am	2007-12-09 20:51:26.000000000 +0300
+++ codeblocks-8.02/src/plugins/contrib/AutoVersioning/Makefile.am	2008-03-02 11:42:41.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libAutoVersioning.la
+plugins_LTLIBRARIES = libAutoVersioning.la
 
 libAutoVersioning_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/BrowseTracker/Makefile.am codeblocks-8.02/src/plugins/contrib/BrowseTracker/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/BrowseTracker/Makefile.am	2008-02-18 23:06:04.000000000 +0300
+++ codeblocks-8.02/src/plugins/contrib/BrowseTracker/Makefile.am	2008-03-02 11:42:57.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libBrowseTracker.la
+plugins_LTLIBRARIES = libBrowseTracker.la
 
 libBrowseTracker_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/byogames/Makefile.am codeblocks-8.02/src/plugins/contrib/byogames/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/byogames/Makefile.am	2007-07-16 18:07:04.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/byogames/Makefile.am	2008-03-02 11:43:56.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libbyogames.la
+plugins_LTLIBRARIES = libbyogames.la
 
 libbyogames_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/cb_koders/Makefile.am codeblocks-8.02/src/plugins/contrib/cb_koders/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/cb_koders/Makefile.am	2007-07-16 18:07:04.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/cb_koders/Makefile.am	2008-03-02 11:45:02.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libcb_koders.la
+plugins_LTLIBRARIES = libcb_koders.la
 
 libcb_koders_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/codesnippets/Makefile.am codeblocks-8.02/src/plugins/contrib/codesnippets/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/codesnippets/Makefile.am	2007-08-08 16:16:12.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/codesnippets/Makefile.am	2008-03-02 11:45:14.000000000 +0300
@@ -7,9 +7,9 @@
         -I$(top_srcdir)/src/include/wxFlatNotebook/include \
         -DBUILDING_PLUGIN
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libcodesnippets.la
+plugins_LTLIBRARIES = libcodesnippets.la
 
 libcodesnippets_la_LDFLAGS = $(WX_GTK2_LIBS) -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/codestat/Makefile.am codeblocks-8.02/src/plugins/contrib/codestat/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/codestat/Makefile.am	2007-03-26 01:31:53.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/codestat/Makefile.am	2008-03-02 11:45:29.000000000 +0300
@@ -4,9 +4,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libcodestat.la
+plugins_LTLIBRARIES = libcodestat.la
 
 libcodestat_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/dragscroll/Makefile.am codeblocks-8.02/src/plugins/contrib/dragscroll/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/dragscroll/Makefile.am	2007-07-16 18:07:04.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/dragscroll/Makefile.am	2008-03-02 11:46:19.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libdragscroll.la
+plugins_LTLIBRARIES = libdragscroll.la
 
 libdragscroll_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/envvars/Makefile.am codeblocks-8.02/src/plugins/contrib/envvars/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/envvars/Makefile.am	2007-07-16 18:07:04.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/envvars/Makefile.am	2008-03-02 11:46:33.000000000 +0300
@@ -2,10 +2,10 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 pkgdata_DATA = envvars.zip
 
-lib_LTLIBRARIES = libenvvars.la
+plugins_LTLIBRARIES = libenvvars.la
 
 libenvvars_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/help_plugin/Makefile.am codeblocks-8.02/src/plugins/contrib/help_plugin/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/help_plugin/Makefile.am	2007-11-15 16:24:03.000000000 +0300
+++ codeblocks-8.02/src/plugins/contrib/help_plugin/Makefile.am	2008-03-02 11:46:55.000000000 +0300
@@ -15,9 +15,9 @@
 		-I$(top_srcdir)/src/include/scripting/sqplus \
 		-include "sdk.h"
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libhelp_plugin.la
+plugins_LTLIBRARIES = libhelp_plugin.la
 
 libhelp_plugin_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/keybinder/Makefile.am codeblocks-8.02/src/plugins/contrib/keybinder/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/keybinder/Makefile.am	2007-07-16 18:07:04.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/keybinder/Makefile.am	2008-03-02 11:47:12.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libkeybinder.la
+plugins_LTLIBRARIES = libkeybinder.la
 
 libkeybinder_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/lib_finder/Makefile.am codeblocks-8.02/src/plugins/contrib/lib_finder/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/lib_finder/Makefile.am	2008-01-02 02:02:41.000000000 +0300
+++ codeblocks-8.02/src/plugins/contrib/lib_finder/Makefile.am	2008-03-02 11:47:25.000000000 +0300
@@ -8,9 +8,9 @@
 		-I$(top_srcdir)/src/include/scripting/squirrel \
 		-I$(top_srcdir)/src/include/scripting/bindings
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = liblib_finder.la
+plugins_LTLIBRARIES = liblib_finder.la
 
 liblib_finder_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/profiler/Makefile.am codeblocks-8.02/src/plugins/contrib/profiler/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/profiler/Makefile.am	2007-03-26 01:31:53.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/profiler/Makefile.am	2008-03-02 11:47:44.000000000 +0300
@@ -4,9 +4,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libProfiler.la
+plugins_LTLIBRARIES = libProfiler.la
 
 libProfiler_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/regex_testbed/Makefile.am codeblocks-8.02/src/plugins/contrib/regex_testbed/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/regex_testbed/Makefile.am	2008-01-13 19:45:24.000000000 +0300
+++ codeblocks-8.02/src/plugins/contrib/regex_testbed/Makefile.am	2008-03-02 11:47:58.000000000 +0300
@@ -2,10 +2,10 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 pkgdata_DATA = RegExTestbed.zip
 
-lib_LTLIBRARIES = libRegExTestbed.la
+plugins_LTLIBRARIES = libRegExTestbed.la
 
 libRegExTestbed_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/source_exporter/Makefile.am codeblocks-8.02/src/plugins/contrib/source_exporter/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/source_exporter/Makefile.am	2007-07-16 18:07:04.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/source_exporter/Makefile.am	2008-03-02 11:48:10.000000000 +0300
@@ -6,9 +6,9 @@
 		-I$(top_srcdir)/src/plugins/contrib/source_exporter/wxPdfDocument/include \
 		-include "sdk.h"
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libexporter.la
+plugins_LTLIBRARIES = libexporter.la
 
 libexporter_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/symtab/Makefile.am codeblocks-8.02/src/plugins/contrib/symtab/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/symtab/Makefile.am	2007-03-26 01:31:53.000000000 +0400
+++ codeblocks-8.02/src/plugins/contrib/symtab/Makefile.am	2008-03-02 11:48:21.000000000 +0300
@@ -4,9 +4,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libSymTab.la
+plugins_LTLIBRARIES = libSymTab.la
 
 libSymTab_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/ThreadSearch/Makefile.am codeblocks-8.02/src/plugins/contrib/ThreadSearch/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/ThreadSearch/Makefile.am	2008-01-13 19:45:24.000000000 +0300
+++ codeblocks-8.02/src/plugins/contrib/ThreadSearch/Makefile.am	2008-03-02 11:43:41.000000000 +0300
@@ -5,9 +5,9 @@
 		-I$(top_srcdir)/src/include/wxscintilla/include \
 		-I$(top_srcdir)/src/include/wxFlatNotebook/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libThreadSearch.la
+plugins_LTLIBRARIES = libThreadSearch.la
 
 libThreadSearch_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/wxSmith/plugin/Makefile.am codeblocks-8.02/src/plugins/contrib/wxSmith/plugin/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/wxSmith/plugin/Makefile.am	2007-11-07 18:26:51.000000000 +0300
+++ codeblocks-8.02/src/plugins/contrib/wxSmith/plugin/Makefile.am	2008-03-02 11:49:17.000000000 +0300
@@ -6,9 +6,9 @@
 		-I$(top_srcdir)/src/include/wxscintilla/include \
 		-I$(top_srcdir)/src/plugins/contrib/propgrid/include
 
-lib_LTLIBRARIES = libwxsmith.la
+plugins_LTLIBRARIES = libwxsmith.la
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
 libwxsmith_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/contrib/wxSmithContribItems/Makefile.am codeblocks-8.02/src/plugins/contrib/wxSmithContribItems/Makefile.am
--- codeblocks-8.02.orig/src/plugins/contrib/wxSmithContribItems/Makefile.am	2008-01-13 19:45:24.000000000 +0300
+++ codeblocks-8.02/src/plugins/contrib/wxSmithContribItems/Makefile.am	2008-03-02 11:49:57.000000000 +0300
@@ -9,9 +9,9 @@
 		-I$(srcdir)/wxchart/wxchart-1.0/include \
 		-I$(srcdir)/wxthings/wxthings/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libwxsmithcontribitems.la
+plugins_LTLIBRARIES = libwxsmithcontribitems.la
 
 libwxsmithcontribitems_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/debuggergdb/Makefile.am codeblocks-8.02/src/plugins/debuggergdb/Makefile.am
--- codeblocks-8.02.orig/src/plugins/debuggergdb/Makefile.am	2007-07-24 18:05:26.000000000 +0400
+++ codeblocks-8.02/src/plugins/debuggergdb/Makefile.am	2008-03-02 11:50:17.000000000 +0300
@@ -7,9 +7,9 @@
 		-I$(top_srcdir)/src/include/scripting/include \
     	-I$(top_srcdir)/src/include/scripting/sqplus
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libdebugger.la
+plugins_LTLIBRARIES = libdebugger.la
 
 libdebugger_la_LDFLAGS = $(WX_GTK2_LIBS) -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/defaultmimehandler/Makefile.am codeblocks-8.02/src/plugins/defaultmimehandler/Makefile.am
--- codeblocks-8.02.orig/src/plugins/defaultmimehandler/Makefile.am	2007-11-15 14:14:48.000000000 +0300
+++ codeblocks-8.02/src/plugins/defaultmimehandler/Makefile.am	2008-03-02 11:50:34.000000000 +0300
@@ -4,9 +4,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libdefaultmimehandler.la
+plugins_LTLIBRARIES = libdefaultmimehandler.la
 
 libdefaultmimehandler_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/openfileslist/Makefile.am codeblocks-8.02/src/plugins/openfileslist/Makefile.am
--- codeblocks-8.02.orig/src/plugins/openfileslist/Makefile.am	2007-11-11 12:24:33.000000000 +0300
+++ codeblocks-8.02/src/plugins/openfileslist/Makefile.am	2008-03-02 11:50:52.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libopenfileslist.la
+plugins_LTLIBRARIES = libopenfileslist.la
 
 libopenfileslist_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/projectsimporter/Makefile.am codeblocks-8.02/src/plugins/projectsimporter/Makefile.am
--- codeblocks-8.02.orig/src/plugins/projectsimporter/Makefile.am	2007-07-10 16:58:12.000000000 +0400
+++ codeblocks-8.02/src/plugins/projectsimporter/Makefile.am	2008-03-02 11:51:01.000000000 +0300
@@ -4,9 +4,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libprojectsimporter.la
+plugins_LTLIBRARIES = libprojectsimporter.la
 
 libprojectsimporter_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/scriptedwizard/Makefile.am codeblocks-8.02/src/plugins/scriptedwizard/Makefile.am
--- codeblocks-8.02.orig/src/plugins/scriptedwizard/Makefile.am	2007-06-20 17:53:35.000000000 +0400
+++ codeblocks-8.02/src/plugins/scriptedwizard/Makefile.am	2008-03-02 11:51:12.000000000 +0300
@@ -6,9 +6,9 @@
 		-I$(top_srcdir)/src/include/scripting/include \
 		-I$(top_srcdir)/src/include/scripting/sqplus
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libscriptedwizard.la
+plugins_LTLIBRARIES = libscriptedwizard.la
 
 libscriptedwizard_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/todo/Makefile.am codeblocks-8.02/src/plugins/todo/Makefile.am
--- codeblocks-8.02.orig/src/plugins/todo/Makefile.am	2007-03-26 01:31:53.000000000 +0400
+++ codeblocks-8.02/src/plugins/todo/Makefile.am	2008-03-02 11:51:22.000000000 +0300
@@ -4,9 +4,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libtodo.la
+plugins_LTLIBRARIES = libtodo.la
 
 libtodo_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
diff --binary -ruN codeblocks-8.02.orig/src/plugins/xpmanifest/Makefile.am codeblocks-8.02/src/plugins/xpmanifest/Makefile.am
--- codeblocks-8.02.orig/src/plugins/xpmanifest/Makefile.am	2007-07-16 18:07:04.000000000 +0400
+++ codeblocks-8.02/src/plugins/xpmanifest/Makefile.am	2008-03-02 11:51:33.000000000 +0300
@@ -2,9 +2,9 @@
 		-I$(top_srcdir)/src/include \
 		-I$(top_srcdir)/src/include/wxscintilla/include
 
-libdir = $(pkgdatadir)/plugins
+pluginsdir = $(pkgdatadir)/plugins
 
-lib_LTLIBRARIES = libxpmanifest.la
+plugins_LTLIBRARIES = libxpmanifest.la
 
 libxpmanifest_la_LDFLAGS = -module -version-info 0:1:0 -shared -no-undefined -avoid-version
 
