--- stage1/Makefile.am.old	2008-11-09 16:50:11.000000000 +0500
+++ stage1/Makefile.am	2008-11-09 16:51:08.000000000 +0500
@@ -12,4 +12,4 @@
 
 SUFFIXES = .exec
 .exec:
-	$(OBJCOPY) -O binary $< $@
+	$(OBJCOPY) -R .note.gnu.build-id -O binary $< $@
--- stage2/Makefile.am.old	2008-11-09 16:52:49.000000000 +0500
+++ stage2/Makefile.am	2008-11-09 16:51:40.000000000 +0500
@@ -269,4 +269,4 @@
 # General rule for making a raw binary.
 SUFFIXES = .exec
 .exec:
-	$(OBJCOPY) -O binary $< $@
+	$(OBJCOPY) -R .note.gnu.build-id -O binary $< $@
