diff -Naur fontconfig-2.6.0-ubuntu-6/conf.d/Makefile.am fontconfig-2.6.0-ubuntu-7/conf.d/Makefile.am
--- fontconfig-2.6.0-ubuntu-6/conf.d/Makefile.am	2009-06-20 12:42:55.000000000 +0200
+++ fontconfig-2.6.0-ubuntu-7/conf.d/Makefile.am	2009-06-20 12:43:15.000000000 +0200
@@ -79,6 +79,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
