diff -Naur fontconfig-2.6.0-ubuntu-3/src/fcname.c fontconfig-2.6.0-ubuntu-4/src/fcname.c
--- fontconfig-2.6.0-ubuntu-3/src/fcname.c	2009-06-20 12:41:09.000000000 +0200
+++ fontconfig-2.6.0-ubuntu-4/src/fcname.c	2009-06-20 12:41:50.000000000 +0200
@@ -440,6 +440,7 @@
     { (FcChar8 *) "lcddefault",	    "lcdfilter",    FC_LCD_DEFAULT },
     { (FcChar8 *) "lcdlight",	    "lcdfilter",    FC_LCD_LIGHT },
     { (FcChar8 *) "lcdlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
+    { (FcChar8 *) "lcdfilterlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
 };
 
 #define NUM_FC_CONSTANTS   (sizeof _FcBaseConstants/sizeof _FcBaseConstants[0])
